src/plugins/GUITestBase/src/tests/common_scenarios/Assembling/extract_consensus/GTTestsAssemblyExtractConsensus.cpp

```diff
@@ -0,0 +1,318 @@
+
+            GTUtilsWizard::clickButton(os, GTUtilsWizard::Run);
+        }
+    };
+
+    const GTLogTracer lt;
+    const QString path1 = testDir + "_common_data/ugenedb/scerevisiae.bam.ugenedb";
+    const QString path2 = dataDir + "samples/Assembly/chrM.sorted.bam";
+
+    //  1. Select "Tools->NGS data analysis->Extract consensus from assemblies..."
+    //  2. Set "_common_data/ugenedb/scerevisiae.bam.ugenedb" and "samples/Assembly/chrM.sorted.bam" as an input
+    //  3. Add "_common_data/bam/small.bam.sorted.bam" to assemblies
+    //  4. Click "Run"
+    GTUtilsDialog::waitForDialog(os, new WizardFiller(
+        os, "Extract Consensus Wizard", new ExtractConsensusWizardWithAddScenario(QStringList() << path1 << path2)
+    ));
+    GTMenu::clickMainMenuItem(os, QStringList() << "Tools"
+                                                << "NGS data analysis"
+                                                << "Extract consensus from assemblies...");
+
+    //  5. Wait for workflow finished
+    GTUtilsTaskTreeView::waitTaskFinished(os, 180000);
+    //  Expected state: There should be no errors in the log
+    CHECK_SET_ERR(!lt.hasErrors(), "Errors in log: " + lt.getJoinedErrorString());
+    //  There should be no notifications in the dashboard
+    CHECK_SET_ERR(!GTUtilsDashboard::hasNotifications(os), "Notifications in dashboard: " +
+        GTUtilsDashboard::getJoinedNotificationsString(os));
+
+    //  6. Return to workflow and call the Extract consensus wizard
+    GTUtilsWorkflowDesigner::returnToWorkflow(os);
+    //  7. Fill in the dialog :
+    //      {Algorithm} SAMtools
+    //      {Keep gaps} False
+    //      {Output file}
+    //     And click "Run"
+    GTUtilsDialog::waitForDialog(os, new WizardFiller(
+        os,
+        "Extract Consensus Wizard",
+        new ExtractConsensusWizardWithAddScenario(QStringList() << path1 << path2, Algo::SAMTOOLS, false, "")
+    ));
+    GTToolbar::clickButtonByTooltipOnToolbar(os, MWTOOLBAR_ACTIVEMDI, "Show wizard");
+
+    //  8. Wait for workflow finished
+    GTUtilsTaskTreeView::waitTaskFinished(os, 180000);
+    //  Expected state: There should be no errors in the log
+    CHECK_SET_ERR(!lt.hasErrors(), "Errors in log: " + lt.getJoinedErrorString());
+    //  There should be no notifications in the dashboard
+    CHECK_SET_ERR(!GTUtilsDashboard::hasNotifications(os), "Notifications in dashboard: " +
+        GTUtilsDashboard::getJoinedNotificationsString(os));
+}
+
+GUI_TEST_CLASS_DEFINITION(wrong_input_test_0001) {
+    const GTLogTracer lt;
+    const QString path1 = dataDir + "samples/Assembly/chrM.fa";
+    const QString path2 = dataDir + "samples/Assembly/chrM.sorted.bam";
+    const QString dashboardErrMsg = "Unsupported document format: ";
+
+    const auto hasDashboardNotification = [](HI::GUITestOpStatus &os, const QString &errMsg) {
+        return !GTWidget::findLabelByText(os, errMsg, GTWidget::findWidget(
```

Comment:
If you will cache results into the local variables you still will have only 3 lines. Split these lines into simpler independent statements.

Reply:
I don't understand what you are talking about

Reply:
Please cache GTWidget::findWidget() result into a separate variable to make the lines 171-173 more readable. I won't increase number of lines you have.